### PR TITLE
Feature: Enhance phone validation

### LINKED ISSUE #837: Feature: Implement libphonenumber-js phone validation
## Context

Following the evaluation in #807 and the approved [research report](https://github.com/kesslerio/attio-mcp-server/blob/main/docs/issues/807-libphonenumber-js-evaluation.md), implement enhanced phone validation using `libphonenumber-js`.

## Objectives

1. Expose `isValidPhoneNumber` validation with structured error messages
2. Implement metadata optimization strategy (min/max split for server/client)
3. Maintain backward compatibility with existing `toE164` API
4. Add comprehensive regression test coverage

## Current State

- **Using**: `parsePhoneNumberFromString` from `libphonenumber-js` in `PhoneNormalizer.ts:6`
- **Gap**: No validation surface (possible vs. valid checks), no granular error messages
- **Issue**: Invalid numbers pass through unchanged in `AttributeAwareNormalizer`

### Phase 1: Core Validation Enhancement
- [ ] Audit current phone utilities (`toE164`, normalizers, validators) - identify all call sites
- [ ] Expose `isValidPhoneNumber` helper in `PhoneNormalizer.ts`
- [ ] Implement structured phone validation errors (e.g., `PhoneValidationError` class)
- [ ] Add `isPossiblePhoneNumber` for lenient validation
- [ ] Update `AttributeAwareNormalizer` to surface validation errors instead of silent pass-through

### Phase 2: Bundle Size Optimization
- [ ] Configure metadata strategy: `min` for CLI/browser, standard for server
- [ ] Update imports to use `libphonenumber-js/min` where appropriate
- [ ] Verify tree-shaking effectiveness with bundle analysis
- [ ] Document bundle size impact in CHANGELOG

### Phase 3: Testing & Validation
- [ ] Add unit tests for new validation helpers (valid, invalid, edge cases)
- [ ] Test international formats, extensions, various country codes
- [ ] Add regression tests covering Issue #798 scenarios
- [ ] Run offline test suite to verify no performance regression
- [ ] Verify bundle size stays within budget (~67 kB min metadata)

### Phase 4: Documentation & Rollout
- [ ] Update developer docs with canonical phone validation patterns
- [ ] Document migration notes in CHANGELOG
- [ ] Update onboarding materials with new validation helpers
- [ ] Add usage examples to README or docs/

### Functionality
- [ ] `isValidPhoneNumber` exposed and returns boolean with validation metadata
- [ ] Structured errors provide actionable feedback (e.g., "Invalid country code", "Too short")
- [ ] `toE164` behavior unchanged (backward compatible)
- [ ] Support for international formats, extensions, and edge cases

### Performance
- [ ] Bundle size CLI/browser: ≤70 kB with min metadata
- [ ] Bundle size server: standard metadata (size unconstrained)
- [ ] No normalization speed regression vs. current implementation
- [ ] Tree-shaking verified (unused helpers excluded)

### Testing
- [ ] Unit test coverage ≥90% for new validation code
- [ ] Regression tests for Issue #798 phone validation scenarios
- [ ] Edge case coverage: extensions, invalid formats, missing country codes
- [ ] Offline test suite passes without API calls

### Documentation
- [ ] Developer docs updated with validation helper usage
- [ ] CHANGELOG documents new capabilities and migration notes
- [ ] Code examples demonstrate valid/invalid/edge case handling
- [ ] Bundle optimization strategy documented

## Technical Notes

**API Design**:
```typescript
// Existing (keep unchanged)
export function toE164(value: string | PhoneNumber): string | null

// New additions
export function isValidPhoneNumber(value: string, country?: string): boolean
export function isPossiblePhoneNumber(value: string, country?: string): boolean
export function validatePhoneNumber(value: string, country?: string): PhoneValidationResult

interface PhoneValidationResult {
  valid: boolean
  possible: boolean
  error?: PhoneValidationError
  e164?: string
}
```

**Metadata Strategy**:
- Server (Node.js): `import { parsePhoneNumber } from 'libphonenumber-js'` (standard metadata)
- CLI: `import { parsePhoneNumber } from 'libphonenumber-js/min'` (~67 kB)
- Browser (future): Lazy-load country-specific metadata or use `min`

**Migration Path**:
1. Implement new helpers alongside `toE164` (no breaking changes)
2. Update `AttributeAwareNormalizer` to use validation helpers
3. Gradually adopt in other services via opt-in
4. No existing data migration needed (E.164 format unchanged)

## Dependencies

- **Blocked by**: None (approved in #807)
- **Blocks**: Issue #798 (phone validation UX improvements)
- **Related**: PR #804 (current phone normalization), PR #831 (research)

## Estimated Effort

**Total**: ~2.25 engineering days (per evaluation)
- Phase 1: 0.5d (core validation)
- Phase 2: 0.5d (bundle optimization)
- Phase 3: 0.5d (testing)
- Phase 4: 0.25d (docs)
- Buffer: 0.5d (integration, unexpected issues)

## Success Metrics

- Phone validation errors provide actionable user feedback
- Bundle size stays within budget (≤70 kB CLI)
- Zero normalization performance regression
- Unblocks Issue #798 phone validation UX work

## References

- Evaluation: docs/issues/807-libphonenumber-js-evaluation.md
- Current impl: src/services/normalizers/PhoneNormalizer.ts
- Library docs: https://www.npmjs.com/package/libphonenumber-js
- Decision comment: https://github.com/kesslerio/attio-mcp-server/issues/807#issuecomment-3363606320

### PR DESCRIPTION
**Pull request by kesslerio**
## Summary
- add central phone validation helpers exposing validatePhoneNumber/isValidPhoneNumber/isPossiblePhoneNumber with structured PhoneValidationError metadata
- route AttributeAwareNormalizer and CLI through the shared helpers, enforcing min metadata bundles for CLI while keeping toE164 backwards compatible
- update docs, changelog, and regression suites to cover actionable errors and metadata strategy for issue #837

## Testing
- npm run test -- --run test/unit/normalizers/phone-number-normalization.test.ts test/utils/phone-validation.test.ts
- npm run test:offline
- npm run build

Resolves #837

### CODE REVIEW

**Comment by kesslerio** [comment]
@claude review @codex review

**Comment by kesslerio** [comment]
🟢 Queued **Sonnet** review in PR-safe context. I'll post results here shortly.

File: src/services/normalizers/AttributeAwareNormalizer.ts [hunk 1]
```diff
@@ -62,11 +139,34 @@ export async function normalizeValues(
 ) {
   void _attributes;
   const out: Record<string, unknown> = { ...values };
+  const phoneIssues: PhoneValidationIssue[] = [];
   for (const [k, v] of Object.entries(values)) {
     const isPhoney = /phone/.test(k); // fast path if you don't want to fetch schemas
     if (isPhoney) {
-      out[k] = normalizePhoneNumbers(v);
+      out[k] = normalizePhoneNumbers(v, k, phoneIssues);
     }
```
[the remark this answers is not in the document]
Comment (kesslerio) [reply src/services/normalizers/AttributeAwareNormalizer.ts:147]: Thanks for flagging. I tightened the guard so normalization now ignores non-string scalars (booleans, flags, etc.) while still running structured validation for real phone payloads. Re-ran the phone normalizer suite to confirm behaviour.

**Comment by kesslerio** [comment]
Addressed the Claude suggestions:

1. Verified the  import via 
> attio-mcp@1.0.0 build
> tsc && tsc-alias


> attio-mcp@1.0.0 postbuild
> shx chmod +x dist/*.js (tsc) – no resolution issues.
2. Added a default-country regression in  to cover national-format inputs.
3. Improved the normalization error messaging to report the number of issues and newline-delimited details, while keeping actionable guidance intact.
4. Documented a concrete  override example in  for workers.

Latest builds: 
> attio-mcp@1.0.0 test
> vitest --run test/utils/phone-validation.test.ts test/unit/normalizers/phone-number-normalization.test.ts


 RUN  v3.2.4 /Users/kesslerio/GDrive/Projects/attio-mcp-server

 ✓ test/utils/phone-validation.test.ts (5 tests) 7ms
 ✓ test/unit/normalizers/phone-number-normalization.test.ts (25 tests) 15ms

 Test Files  2 passed (2)
      Tests  30 passed (30)
   Start at  21:50:12
   Duration  627ms (transform 205ms, setup 386ms, collect 79ms, tests 22ms, environment 0ms, prepare 255ms), the offline equivalents, and 
> attio-mcp@1.0.0 build
> tsc && tsc-alias


> attio-mcp@1.0.0 postbuild
> shx chmod +x dist/*.js.

**Comment by kesslerio** [comment]
@claude re-review

**Comment by kesslerio** [comment]
🔁 Re-queued **Sonnet** review in PR-safe context. I'll post results here shortly.

**Comment by kesslerio** [comment]
Follow-up on latest Claude review:

- Added a true single-source export: `PhoneNormalizer.ts` now purely re-exports the helpers (including a shared `toE164`) from `phone-validation.ts`, avoiding wrapper duplication while staying backward compatible.
- Kept the aggregated phone validation message but now tag multi-error cases with the generic `phone_numbers` field. (Full per-error metadata isn’t supported by `UniversalValidationError` today, but the newline-delimited details call out each failing path.)
- PR description already includes `Resolves #837`; no further change needed per the workflow policy.

Validation rerun:
- `npm run test -- --run test/utils/phone-validation.test.ts test/unit/normalizers/phone-number-normalization.test.ts`
- `npm run test:offline -- --run test/utils/phone-validation.test.ts test/unit/normalizers/phone-number-normalization.test.ts`
- `npm run build`
